schema/object.go

```diff
@@ -684,6 +652,44 @@ func (a *AnyTypedObject[T]) Any() TypedObject[any] {
 	return a
 }
 
+// ConvertToObjectSchema attempts to extract an ObjectSchema from the input.
+//
+// If an ObjectSchema is found, it returns it.
+// If a RefSchema is found, it extracts the cached object schema the ref is referencing.
+// If a ScopeSchema is found, it extracts the root object schema.
+// Returns the ObjectSchema and true if successful, otherwise nil and false.
+func ConvertToObjectSchema(typeOrData any) (Object, bool) {
+	// Try plain object schema
+	objectSchemaType, ok := typeOrData.(*ObjectSchema)
+	if ok {
+		return objectSchemaType, true
+	}
+	// Next, try ref schema
+	refSchemaType, ok := typeOrData.(*RefSchema)
+	if ok {
+		return refSchemaType.GetObject(), true
+	}
+	// Next, try scope schema.
+	scopeSchemaType, ok := typeOrData.(*ScopeSchema)
+	if ok {
+		return scopeSchemaType.Objects()[scopeSchemaType.Root()], true
+	}
```

Comment:
This seems like it should be a `switch` statement, no?
```go
	switch v := typeOrData.(type) {
	case *ObjectSchema:
		return v, true
	case *RefSchema:
		return v.GetObject(), true
	case *ScopeSchema:
		return v.Objects()[scopeSchemaType.Root()], true
	}
```

Reply:
It's worth trying. If it fails, lots of tests for ValidateCompatibility will fail.